src/pkg/utils/bytes.go

```diff
@@ -50,3 +55,35 @@ func ByteFormat(inputNum float64, precision int) string {
 
 	return strconv.FormatFloat(returnVal, 'f', precision, 64) + unit
 }
+
+// CheckLocalProgress continuously checks the local directory size and updates the progress bar.
+// NOTE: This function runs infinitely until the completeChan is triggered.
+func CheckLocalProgress(progressBar *message.ProgressBar, expectedTotal int64, filepath string, wg *sync.WaitGroup, completeChan chan int, updateText string) {
```

Comment:
As mentioned this should probably be moved to avoid the utils import of message.go